### PR TITLE
Initial specs.

### PR DESCRIPTION
**Pull request by Zambelli24**

### CODE REVIEW

File: src/clashboard/clash_interface.py [hunk 1]
```diff
@@ -6,20 +6,38 @@ def __init__(self, db):
...
+        :param filter_name: the specific value to filter on in that column
+        """
+        pass
 
     # Uncomment when tested and implemented
-    # def apply_filter(self, filter):
-    #    pass
+    def apply_filter(self, filter_category, filter_name):
+        """
+        Adds the specified filter to the list of filters, and re-runs the current query
```
Comment (AndCarr) [inline src/clashboard/clash_interface.py]: Perhaps "modified query" instead of "current query"

File: src/clashboard/clash_interface.py [hunk 2]
```diff
@@ -6,20 +6,38 @@ def __init__(self, db):
...
+        :return: the list of human-readable strings
+        """
+        pass
 
     # Uncomment when tested and implemented
-    # def set_group_by(self, attribute):
-    #    pass
+    def set_group_by(self, attribute):
+        """
+        Sets the attribute to group the data by, runs the query for the first
```
Comment (AndCarr) [inline src/clashboard/clash_interface.py]: "for the first" what?

**Comment by Charles-Peeke** [comment]
👍

**Comment by jmadilia** [comment]
👌

**Comment by ddurski22** [comment]
Approved

**Comment by ROCCOF1** [comment]
Approve

**Pull request merged**